src/errors.ts

```diff
@@ -57,10 +63,24 @@ export class HttpError extends RequestError {
   }
 }
 
+function isAbortError(err: unknown): err is DOMException {
+  return Boolean(err && err instanceof Error && err.name === 'AbortError');
+}
+
 export default function errorFactory(err: unknown) {
+  if (isAbortError(err)) {
+    return new RequestCanceledError(
```

Comment:
After this change, this is no longer a general error factory as it converts abort error to the `RequestCanceledError`. Please adjust the name of the function.